src/lib/util/corefile.cpp

```diff
@@ -1270,11 +1270,11 @@ core_file::core_file()
     assumptions about path separators
 -------------------------------------------------*/
 
-std::string core_filename_extract_base(const char *name, bool strip_extension)
+std::string core_filename_extract_base(const std::string &name, bool strip_extension)
 {
 	/* find the start of the name */
-	const char *start = name + strlen(name);
-	while (start > name && !util::is_directory_separator(start[-1]))
```

Comment:
Why use `c_str` and work with pointers and a manual loop when you can use reverse iterators and `find_if`?  This is the kind of C to C++ conversion that minimises actual benefits.